spring-cloud-starter-stream-sink-jdbc/src/test/java/org/springframework/cloud/stream/app/jdbc/sink/JdbcSinkIntegrationTests.java

```diff
@@ -76,6 +76,39 @@ public void testInsertion() {
 		}
 
 	}
+	
+  @TestPropertySource(properties = "jdbc.batchSize=1000")
+	public static class SimpleBatchInsertTests extends JdbcSinkIntegrationTests {
+
+		@Test
+		public void testBatchInsertion() {
+      final int numberOfInserts = 5000;
+			Payload sent = new Payload("hello", 42);
+      for (int i = 0; i < numberOfInserts; i++) {
+  			channels.input().send(MessageBuilder.withPayload(sent).build());
+      }
+      int result = jdbcOperations.queryForObject("select count(*) from messages", Integer.class);
+			Assert.assertThat(result, is(numberOfInserts));
+		}
+
+	}
+
+  @TestPropertySource(properties = { "jdbc.batchSize=1000", "jdbc.idleTimeout=1000" })
+	public static class BatchInsertTimeoutTests extends JdbcSinkIntegrationTests {
+
+		@Test
+		public void testBatchInsertionTimeout() throws InterruptedException {
+      final int numberOfInserts = 10;
+			Payload sent = new Payload("hello", 42);
+      for (int i = 0; i < numberOfInserts; i++) {
+  			channels.input().send(MessageBuilder.withPayload(sent).build());
+      }
+      Assert.assertThat(jdbcOperations.queryForObject("select count(*) from messages", Integer.class), is(0));
+      Thread.sleep(2000); // wait 2s
```

Comment:
May we don't sleep so long?
Why just simple `jdbc.batchSize=10000000` and `jdbc.idleTimeout=1` are not enough for us to achieve expectations in the test?

Reply:
Sure, that makes sense. I reduced `jdbc.idleTimeout` to 100 ms and the test wait time to 200 ms. Much shorter times seem to cause the test to fail occasionally, as the embedded database used for testing runs asynchronously. See commit 2414ba2a1ccca536686c1892ef2a88a4de53120c .